elastic/eui
# [Tech debt] Create new CHANGELOG.md workflow that avoids merge conflicts between PRs

## Summary

This PR changes our changelog workflow. Instead of editing `CHANGELOG.md` directly, each PR should create a new file in a new `upcoming_changelogs/` directory titled by the ID of the PR, e.g. `5555.md`. The content itself remains mostly the same before, but separate files brings us the following benefits:

- No git merge conflicts when another PR merges in its changelog changes ahead of yours
- No changelogs going to the wrong release header if someone forgets to fix their changelog after a release happens
- No time-sensitive hour block while a release is happening: if a new PR merges in accidentally while a release is running, the new file just stays in main and isn't considered part of the release.
- No need to copy and paste a PR link repeatedly if your changelog has multiple bullet points - the script will handle automatically appending a PR link (based on the file name)

## QA

`npm run release --dry-run` doesn't appear to work as expected but I ran it as far as I dared and I'm relatively confident this is working.

Here's the output console log of the updated version check:

<img width="690" alt="" src="https://user-images.githubusercontent.com/549407/158690306-b0495b61-75dd-4742-8dbe-05483c97151d.png">

Here's the output of the new script/regexes based on the changelogs in this PR:

```
## [`52.0.0`](https://github.com/elastic/eui/tree/v52.0.0)

- Added `editorChecklist` glyph to `EuiIcon` ([#5705](https://github.com/elastic/eui/pull/5705))
- Updated `testenv` mock for `EuiIcon` to render `aria-label` as text ([#5709](https://github.com/elastic/eui/pull/5709))
- Added `compressed` prop to `EuiFilterGroup` and reduced the size of the `EuiFilterButton` notification badge ([#5717](https://github.com/elastic/eui/pull/5717))

**Breaking changes**

- Removed Legacy theme including compiled CSS ([#5688](https://github.com/elastic/eui/pull/5688))
- Removed `flush` and `size` props in `EuiFilterButtonProps` ([#5717](https://github.com/elastic/eui/pull/5717))

**CSS-in-JS conversions**

- Converted `EuiMark` to CSS-in-JS styling ([#4575](https://github.com/elastic/eui/pull/4575))
```

It would be great to guinea pig test this PR on this week's release! 🤠 

### Checklist
